lib/data/external/GcpClient.js

```diff
@@ -77,6 +87,202 @@ class GcpClient extends AwsClient {
+            if (isNaN(number)) {
+                return callback(errors.MalformedXML);
+            }
+            const partNumber = parseInt(number, 10);
+            const partParams = {
+                PartName: createMpuKey(gcpKey, uploadId, partNumber),
+                PartNumber: partNumber,
+                ETag: partObj.ETag[0],
+            };
+            partArray.push(partParams);
+        }
+        const mpuParams = {
+            Bucket: this._gcpBucketName,
+            MPU: this._mpuBucketName,
+            Key: gcpKey,
+            UploadId: uploadId,
+            MultipartUpload: { Parts: partArray },
+        };
+        const completeObjData = { key: gcpKey };
+        return this._client.completeMultipartUpload(mpuParams,
+        (err, completeMpuRes) => {
+            if (err) {
+                logHelper(log, 'error', 'err from data backend on ' +
+                'completeMPU', err, this._dataStoreName, this.clientType);
+                return callback(errors.ServiceUnavailable
+                  .customizeDescription('Error returned from ' +
+                  `GCP: ${err.message}`)
+                );
+            }
+            if (!completeMpuRes.VersionId) {
+                logHelper(log, 'error', 'missing version id for data ' +
+                'backend object', missingVerIdInternalError,
+                    this._dataStoreName, this.clientType);
+                return callback(missingVerIdInternalError);
+            }
+            completeObjData.eTag = removeQuotes(completeMpuRes.ETag);
+            completeObjData.dataStoreVersionId = completeMpuRes.VersionId;
+            completeObjData.contentLength = completeMpuRes.ContentLength;
+            return callback(null, completeObjData);
+        });
+    }
+
+    uploadPart(request, streamingV4Params, stream, size, key, uploadId,
+    partNumber, bucketName, log, callback) {
+        let hashedStream = stream;
+        if (request) {
+            const partStream = prepareStream(request, streamingV4Params,
+                log, callback);
+            hashedStream = new MD5Sum();
+            partStream.pipe(hashedStream);
+        }
+
+        const gcpKey = this._createGcpKey(bucketName, key, this._bucketMatch);
+        const params = {
+            Bucket: this._mpuBucketName,
+            Key: gcpKey,
+            UploadId: uploadId,
+            Body: hashedStream,
+            ContentLength: size,
+            PartNumber: partNumber };
+        return this._client.uploadPart(params, (err, partResObj) => {
+            if (err) {
+                logHelper(log, 'error', 'err from data backend ' +
+                  'on uploadPart', err, this._dataStoreName, this.clientType);
+                return callback(errors.ServiceUnavailable
+                  .customizeDescription('Error returned from ' +
+                  `GCP: ${err.message}`)
+                );
+            }
+            const noQuotesETag = removeQuotes(partResObj.ETag);
```

Comment:
I think we need a comment everywhere this used to indicate why the quotes are being removed. Just a short description would be enough.

Reply:
will add the comments.